flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannelTest.java

```diff
@@ -170,6 +170,65 @@ public Void call() throws Exception {
 		}
 	}
 
+	@Test
+	public void testConcurrentNotifyBufferAvailableAndRelease() throws Exception {
+		// Config
+		// Repeatedly spawn two tasks: one to notify buffer availability and the other to release the channel
+		// concurrently. We do this repeatedly to provoke races.
+		final int numberOfRepetitions = 1024;
+
+		// Setup
+		final ExecutorService executor = Executors.newFixedThreadPool(2);
+		final Buffer buffer = TestBufferFactory.createBuffer(TestBufferFactory.BUFFER_SIZE);
+
+		try {
+			// Test
+			final SingleInputGate inputGate = mock(SingleInputGate.class);
```

Comment:
Please no mockito in new tests. Use real object or create proper mock.

Reply:
This is outdated(?) - there is no new test code in that regard - only old code that is made more generic ;)
-> is that enough?